src/main/java/com/shopify/store/Discount.java

```diff
@@ -0,0 +1,18 @@
+package com.shopify.store;
+
+import java.math.BigDecimal;
+
+/**
+ * Discount uses different Discount strategies
+ */
+public class Discount {
+    private DiscountStrategy strategy;
+
+    public Discount(DiscountStrategy strategy){
+        this.strategy = strategy;
+    }
+
+    public BigDecimal applyDiscount(long price, int quantity, double percentage){
```

Comment:
A `line item` abstraction would be worthwhile, so you don't pass around quantity and price directly. 

A `LineItem` could involve a `Product` which has a `name` and a `unitPrice`. You could then make a `DiscountRegistry` that associated `Product` objects to `DiscountStrategy` objects.